pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -1952,6 +1952,7 @@ which makes the code also more readable.
                 <value>
                     <![CDATA[
 //VariableInitializer
+/../..[(Type)]
 //PrimaryExpression[not(PrimarySuffix)]
```

Comment:
You can probably replace `//` with `/Expression/`, so as not to recurse on all descendants. I think then you wouldn't need the `[not(ancestor::ArgumentList)]` predicate. I have not tested it on the test cases though

In general using `//` anywhere else than on the first path segment can make a rule horribly inefficient

Reply:
`List<String> x = new ArrayList<String>()` will not be detected